File: serializers/gson/src/main/java/at/stefangeyer/challonge/serializer/gson/adapter/TournamentQueryAdapter.java
package at.stefangeyer.challonge.serializer.gson.adapter;

import at.stefangeyer.challonge.model.query.TournamentQuery;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonSerializationContext;
import com.google.gson.JsonSerializer;

import java.lang.reflect.Type;

public class TournamentQueryAdapter implements JsonSerializer<TournamentQuery> {

    @Override
    public JsonElement serialize(TournamentQuery src, Type typeOfSrc, JsonSerializationContext context) {
        JsonObject entity = new JsonObject();

        if (src.getName() != null) {
            entity.addProperty("name", src.getName());
        }

        if (src.getUrl() != null) {
            entity.addProperty("url", src.getUrl());
        }

        if (src.getTournamentType() != null) {
            entity.addProperty("tournament_type", src.getTournamentType().toString().toLowerCase().replace("_", " "));
        }

        if (src.getSubdomain() != null) {
            entity.addProperty("subdomain", src.getSubdomain());
        }

        if (src.getDescription() != null) {
            entity.addProperty("description", src.getDescription());
        }

        if (src.getOpenSignup() != null) {
            entity.addProperty("open_signup", src.getOpenSignup());
        }

        if (src.getHoldThirdPlaceMatch() != null) {
            entity.addProperty("hold_third_place_match", src.getHoldThirdPlaceMatch());
        }

        if (src.getPointsForMatchWin() != null) {
            entity.addProperty("pts_for_match_win", src.getPointsForMatchWin());
        }

        if (src.getPointsForMatchTie() != null) {
            entity.addProperty("pts_for_match_tie", src.getPointsForMatchTie());
        }

        if (src.getPointsForGameWin() != null) {
            entity.addProperty("pts_for_game_win", src.getPointsForGameWin());
        }

        if (src.getPointsForGameTie() != null) {
            entity.addProperty("pts_for_game_tie", src.getPointsForGameTie());
        }

        if (src.getPointsForBye() != null) {
            entity.addProperty("pts_for_bye", src.getPointsForBye());
        }

        if (src.getSwissRounds() != null) {
            entity.addProperty("swiss_rounds", src.getSwissRounds());
        }

        if (src.getRankedBy() != null) {
            entity.addProperty("ranked_by", src.getRankedBy().toString().toLowerCase().replace("_", " "));
        }

        if (src.getRoundRobinPointsForGameWin() != null) {
            entity.addProperty("rr_pts_for_game_win", src.getRoundRobinPointsForGameWin());
        }

        if (src.getRoundRobinPointsForGameTie() != null) {
            entity.addProperty("rr_pts_for_game_tie", src.getRoundRobinPointsForGameTie());
        }

        if (src.getRoundRobinPointsForMatchWin() != null) {
            entity.addProperty("rr_pts_for_match_win", src.getRoundRobinPointsForMatchWin());
        }

        if (src.getRoundRobinPointsForMatchTie() != null) {
            entity.addProperty("rr_pts_for_match_tie", src.getRoundRobinPointsForMatchTie());
        }

        if (src.getAcceptAttachments() != null) {
            entity.addProperty("accept_attachments", src.getAcceptAttachments());
        }

        if (src.getHideForum() != null) {
            entity.addProperty("hide_forum", src.getHideForum());
        }

        if (src.getShowRounds() != null) {
            entity.addProperty("show_rounds", src.getShowRounds());
        }

        if (src.getPrivateOnly() != null) {
            entity.addProperty("private", src.getPrivateOnly());
        }

        if (src.getNotifyUsersWhenMatchesOpen() != null) {
            entity.addProperty("notify_users_when_matches_open", src.getNotifyUsersWhenMatchesOpen());
        }

        if (src.getNotifyUsersWhenTheTournamentEnds() != null) {
            entity.addProperty("notify_users_when_the_tournament_ends", src.getNotifyUsersWhenTheTournamentEnds());
        }

        if (src.getSequentialPairings() != null) {
            entity.addProperty("sequential_pairings", src.getSequentialPairings());
        }

        if (src.getSignupCap() != null) {
            entity.addProperty("signup_cap", src.getSignupCap());
        }

        if (src.getStartAt() != null) {
            entity.add("start_at", context.serialize(src.getStartAt()));
        }

        if (src.getCheckInDuration() != null) {
            entity.addProperty("check_in_duration", src.getCheckInDuration());
        }

        if (src.getGrandFinalsModifier() != null) {
            entity.addProperty("grand_finals_modifier", src.getGrandFinalsModifier().toString().toLowerCase().replace("_", " "));
        }

        if (src.getTieBreaks() != null) {
            entity.add("tie_breaks", context.serialize(src.getTieBreaks()));
        }

        return entity;
    }
}

File: src/main/kotlin/at/stefangeyer/challonge/serializer/gson/adapter/TournamentQueryAdapter.kt
/*
 * Copyright 2017 Stefan Geyer <stefangeyer at outlook.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package at.stefangeyer.challonge.serializer.gson.adapter

import at.stefangeyer.challonge.model.query.TournamentQuery
import com.google.gson.*
import java.lang.reflect.Type

class TournamentQueryAdapter internal constructor() : JsonSerializer<TournamentQuery> {

    override fun serialize(query: TournamentQuery, type: Type, context: JsonSerializationContext): JsonElement {
        val parent = JsonObject()

        val tqEntity = JsonObject()

        if (query.name != null) tqEntity.addProperty("name", query.name)
        if (query.url != null) tqEntity.addProperty("url", query.url)
        if (query.tournamentType != null) tqEntity.addProperty("tournament_type", query.tournamentType.toString().toLowerCase().replace("_", " "))
        if (query.subdomain != null) tqEntity.addProperty("subdomain", query.subdomain)
        if (query.description != null) tqEntity.addProperty("description", query.description)
        if (query.openSignup != null) tqEntity.addProperty("open_signup", query.openSignup)
        if (query.holdThirdPlaceMatch != null) tqEntity.addProperty("hold_third_place_match", query.holdThirdPlaceMatch)
        if (query.pointsForMatchWin != null) tqEntity.addProperty("pts_for_match_win", query.pointsForMatchWin)
        if (query.pointsForMatchTie != null) tqEntity.addProperty("pts_for_match_tie", query.pointsForMatchTie)
        if (query.pointsForGameWin != null) tqEntity.addProperty("pts_for_game_win", query.pointsForGameWin)
        if (query.pointsForGameTie != null) tqEntity.addProperty("pts_for_game_tie", query.pointsForGameTie)
        if (query.pointsForBye != null) tqEntity.addProperty("pts_for_bye", query.pointsForBye)
        if (query.swissRounds != null) tqEntity.addProperty("swiss_rounds", query.swissRounds)
        if (query.rankedBy != null) tqEntity.addProperty("ranked_by", query.rankedBy.toString().toLowerCase().replace("_", " "))
        if (query.roundRobinPointsForGameWin != null) tqEntity.addProperty("rr_pts_for_game_win", query.roundRobinPointsForGameWin)
        if (query.roundRobinPointsForGameTie != null) tqEntity.addProperty("rr_pts_for_game_tie", query.roundRobinPointsForGameTie)
        if (query.roundRobinPointsForMatchWin != null) tqEntity.addProperty("rr_pts_for_match_win", query.roundRobinPointsForMatchWin)
        if (query.roundRobinPointsForMatchTie != null) tqEntity.addProperty("rr_pts_for_match_tie", query.roundRobinPointsForMatchTie)
        if (query.acceptAttachments != null) tqEntity.addProperty("accept_attachments", query.acceptAttachments)
        if (query.hideForum != null) tqEntity.addProperty("hide_forum", query.hideForum)
        if (query.showRounds != null) tqEntity.addProperty("show_rounds", query.showRounds)
        if (query.private != null) tqEntity.addProperty("private", query.private)
        if (query.notifyUsersWhenMatchesOpen != null) tqEntity.addProperty("notify_users_when_matches_open", query.notifyUsersWhenMatchesOpen)
        if (query.notifyUsersWhenTheTournamentEnds != null) tqEntity.addProperty("notify_users_when_the_tournament_ends", query.notifyUsersWhenTheTournamentEnds)
        if (query.sequentialPairings != null) tqEntity.addProperty("sequential_pairings", query.sequentialPairings)
        if (query.signupCap != null) tqEntity.addProperty("signup_cap", query.signupCap)
        if (query.startAt != null) tqEntity.add("start_at", context.serialize(query.startAt))
        if (query.checkInDuration != null) tqEntity.addProperty("check_in_duration", query.checkInDuration)
        if (query.grandFinalsModifier != null) tqEntity.addProperty("grand_finals_modifier", query.grandFinalsModifier.toString().toLowerCase().replace("_", " "))
        tqEntity.add("tie_breaks", context.serialize(query.tieBreaks))

        parent.add("tournament", tqEntity)

        return parent
    }
}
